### PR TITLE
siteaccess aware config

### PR DESCRIPTION
**Pull request by bjko**
make config for workflow_directory siteaccess aware to have different workflow definitions for different sitaccess settings, eg:

```
ez_workflow_engine:
    system:
        SITEACCESSNAME:
            workflow_directory: 'WorkflowDefinitions_what_ever'
```

### CODE REVIEW

**Comment by gggeek** [comment]
Seems like good work, thanks!

I just have a question about the split of the parameters in 2 yaml files: it seems that all of parameters are now defined in both places ?

**Comment by bjko** [comment]
Not accactly. First we had plans to do this for all of the 4 settings, but not enough time. With this pr only the directory is sitaccess aware. The not used settings are commeted out. If we had some more time in future, we will do this with a new pr and delete the old settings.

**Comment by bjko** [comment]
Do you need any futher information or assistance?

**Comment by gggeek** [comment]
Actually I am myself in a similar situation right now, but for migration bundle (one ez install with 2 dbs for different sites, need to use a different set of migrations for each db), so I am looking at where/how to implement this best.

Also I want to check how this behaves wrt the 'suspend/resume' and 'status' commands...

File: DependencyInjection/EzWorkflowEngineExtension.php [hunk 1]
```diff
@@ -15,14 +17,27 @@
...
-        $config = $this->processConfiguration($configuration, $configs);
+        $config        = $this->processConfiguration( $configuration, $configs );
+
+        $loader = new Loader\YamlFileLoader( $container, new FileLocator( __DIR__ . '/../Resources/config' ) );
+        $loader->load( 'services.yml' );
+        $loader->load( 'default_settings.yml' );
 
-        $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
-        $loader->load('services.yml');
+        $processor = new ConfigurationProcessor( $container, 'ez_workflowengine_bundle' );
```
Comment (gggeek) [inline DependencyInjection/EzWorkflowEngineExtension.php:37]: I am not sure that I fully understand this.
I my mind, either we have the users put parameters in config files (1), or we have the users put semantic-config in config files, and then transform that config into services or parameters (2).

It seems here that you went for approach 1, which should work well enough for the specified needs (having the config change per-siteaccess), as long as users use the correct format for the parameter name.

The usage of the ConfigurationProcessor instead seems to point to option 2, where we want the users to define the setting using semantic-configuration format, and have this class create the value of the parameter.
But in this case:
- why not document the available semantic config and how to set it up ?
- why create the default_settings.yml file ?

**Comment by gggeek** [comment]
btw: slightly different (simpler ?) approach taken in mig bundle 4.7.0, released today. Will adopt the same for workflow bundle.

**Comment by gggeek** [comment]
Implemented (slightly differently) in 1.1. Thx for the idea!

**Pull request closed without merging**